### ISSUE TITLE
Cannot deploy Firebase Cloud Functions with current billing plan

### ISSUE THREAD

**Issue by H-Yulong** [homotopy-io/homotopy-rs#842]
Google does not allow users to deploy cloud functions with the free plan (Spark, which we're using currently). We must upgrade to a pay-as-you-go plan (Blaze) to use cloud functions. 

For pricing, see [here](https://firebase.google.com/pricing) for full details. Basically, we can still deploy functions for free within a certain amount of usage in the Blaze plan, but Google charges hosting, database, and cloud storage for a few cents per GB.

(I can still develop and test project CRUD functionalities with local firebase emulators though)

**Comment by jamievicary** [comment]
Can you estimate realistic price per year?

**Comment by jamievicary** [comment]
For our use case

**Comment by calintat** [comment]
Were we not using cloud functions in the JS version? Or did they start charging for them since then?

**Comment by H-Yulong** [comment]
@calintat Yes, I think Google start charging for cloud functions in late 2020.

@jamievicary The estimated price for a small app (with 50k installs and 5k daily users) is $12.14/month, as [this page](https://firebase.google.com/docs/firestore/billing-example#small-50k-installs) suggests. The price could be much lower (perhaps less than $1.00/month) since we don't really have many users.

**Comment by jamievicary** [comment]
Can you explain why it would be difficult? I thought we just used a simple json data structure.

**Comment by H-Yulong** [comment]
> Can you explain why it would be difficult? I thought we just used a simple json data structure.

Sorry for not being clear -- what I mean is that using some free alternative to Firebase would be difficult. We need to dive into another framework and start over.

**Comment by jamievicary** [comment]
Ok but that issue is not about the data, it is about switching to a different API, right?

**Comment by H-Yulong** [comment]
> Ok but that issue is not about the data, it is about switching to a different API, right?

Yes, it's about switching to another API.

**Comment by NickHu** [comment]
It is still free, you just need to be on their pay-as-you-go plan: https://firebase.google.com/pricing
They have a free monthly allowance that I doubt we'll ever exceed.

**Comment by jamievicary** [comment]
That looks good to me, does everyone agree?
